physionet-django/project/models.py

```diff
@@ -1533,28 +1533,179 @@ def get_storage_info(self, force_calculate=True):
+            Chicago : Chicago (17th edition)
+            Harvard : Harvard
+            Vancouver : Vancouver
+
+        Returns
+        -------
+        citation_dict (dictionary) : dictionary of each citation style and
+            their respective citation for the requested project
+        """
+        # For if there are no authors or valuable metadata
         if self.is_legacy:
-            return ''
+            return {'APA':''}
 
+        styles = ['APA','MLA','Chicago','Harvard','Vancouver']
         authors = self.authors.all().order_by('display_order')
-        if self.doi:
-            return format_html(
-                '{authors} ({year}). {title} (version {version}). '
-                '<i>PhysioNet</i>. '
-                '<a href="https://doi.org/{doi}">https://doi.org/{doi}</a>',
-                authors=', '.join(a.initialed_name() for a in authors),
-                year=self.publish_datetime.year,
-                title=self.title,
-                version=self.version,
-                doi=self.doi)
-        else:
-            return format_html(
-                '{authors} ({year}). {title} (version {version}). '
-                '<i>PhysioNet</i>.',
-                authors=', '.join(a.initialed_name() for a in authors),
-                year=self.publish_datetime.year,
-                title=self.title,
-                version=self.version)
+        citation_dict = {}
+
+        for style in styles:
+
+            if (style == 'MLA'):
+
+                if (len(authors) == 1):
+                    temp_authors = authors[0].get_full_name(reverse=True)
+                elif (len(authors) == 2):
+                    first_author = authors[0].get_full_name(reverse=True)
+                    second_author = authors[1].get_full_name()
+                    temp_authors = first_author + ', and ' + second_author
+                else:
+                    temp_authors = ', '.join(authors[0].get_full_name().split())
+                    temp_authors += ', et al'
+
+                if self.doi:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}), doi:'
+                        '<a href="https://doi.org/{doi}">https://doi.org/{doi}</a>.',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year,
+                        doi=self.doi)
+                else:
+                    citation_format = format_html(
+                        '{authors}. "{title} (version {version})." '
+                        '<i>PhysioNet</i> ({year}).',
+                        authors=temp_authors,
+                        title=self.title,
+                        version=self.version,
+                        year=self.publish_datetime.year)
+
+            elif (style == 'APA'):
```

Comment:
no need for the parentheses in these clauses `style == 'APA:` is fine

Reply:
Sorry, just habit since it was always easier for me to read logically but I will change it.